configure.py

```diff
@@ -2175,4 +2176,7 @@ def make_bin_override():
 
 print_verbose("running: \n    " + " ".join(['python', 'tools/gyp_node.py'] + gyp_args))
 run_gyp(gyp_args)
+if sys.platform == 'win32':
+  os.path.isfile('./compile_commands.json') and os.unlink('./compile_commands.json')
+  shutil.copy2('./out/' + config['BUILDTYPE'] + '/compile_commands.json', './compile_commands.json')
```

Comment:
```suggestion
  shutil.copy2('./' + config['BUILDTYPE'] + '/compile_commands.json', './compile_commands.json')
```

I've tested this change (with https://github.com/nodejs/gyp-next/pull/238 included) and saw that this line was failing. The reason was the wrong src path here. With this change I got project generation to suceed.

Reply:
Strange, isn't `./{Debug,Release}` just a junction for `./out/{Debug,Release}`...?

Reply:
Yes, it is, but in `vcbuild.bat` it is created [after the compilation](https://github.com/nodejs/node/blob/9efc84a2cb32af325f06c465ba82175b67089fe0/vcbuild.bat#L344) so it doesn't exist at this time

Reply:
Yes. But this change switches it to look for the symlink instead of the out folder which is the real folder no?

Reply:
My mistake, I didn't explain well...

Yes, you are right Release and Debug folders are just junctions to out/Release and out/Debug. However, using the new option eg. `vcbuild.bat compile-commands` will create real folders Debug and Release as part of the configuration step, before compilation.

Reply:
One more very important detail - since using `compile-commands` results in folders Debug and Release being created with compile_commands.json inside, [the line I referenced earlier](https://github.com/nodejs/node/blob/9efc84a2cb32af325f06c465ba82175b67089fe0/vcbuild.bat#L344) will fail thus failing the entire build with this message:
```
The directory is not empty.
Cannot create a file when that file already exists.
"Could not create junction to 'out\Release'."
```
The way things are looking, after copying the selected build type's JSON to root, both of the originals will have to be deleted.

Reply:
Hmm or perhaps we want to fix things so that the `compile_commands.json` is written into `out/{Debug,Release}` like it does on Unix... Not sure why it would create those folders and write it there...

Reply:
I think the reason for that can be found in the PR you've landed previously https://github.com/nodejs/gyp-next/pull/238.

```
output_dir = params["options"].generator_output or os.getcwd()
```
This will be cwd on Windows, so
```
filename = os.path.join(output_dir, configuration_name, "compile_commands.json")
```
will be in `{PROJECT_ROOT}\{Release|Debug}\compile_commands.json`

Reply:
Yeah... figures... I guess a more correct fix might be to fallback to the code in the else if `generator_output` is `None`...

Reply:
Submitted https://github.com/nodejs/gyp-next/pull/239

Reply:
Is the current implementation correct @StefanStojanovic?